src/compiler/scale.ts

```diff
@@ -41,7 +41,7 @@ export function compileScales(names: Array<string>, model: Model) {
   }, []);
 }
 
-export function type(channel: Channel, model: Model) {
+export function type(channel: Channel, model: Model): Type | string {
```

Comment:
this returns scale’s type, not the data `Type`, so only `string` is the correct semantic.